simplify and update development plan (#306)
icesat2py/icepyx: doc/source/contributing/development_plan.rst

icepyx Development Plan
=======================

> This page provides a high-level overview of some focus areas of ongoing and for future icepyx development.

Reviewer: ```suggestion
This page provides a high-level overview of focus areas for icepyx's ongoing and future development.
```

> This list is not exclusive or highly specific.

Author: How about: "This list is intentionally general and not exhaustive." Clearer or needs work?

Instead, specific development tasks and new functionality implementations are driven by individual developers/teams.

Our ongoing efforts are tracked as issues on our GitHub `issue tracker <https://github.com/icesat2py/icepyx/issues>`_.
We invite you to join the active discussions happening there.

Major Themes for Development
----------------------------

Enhancing User Interactivity and Visualization
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
icepyx aims to continually reduce the need for researchers to rewrite "routine" code by enabling easy visualization

> throughout the data inquiry to analyzed data presentation process and provide a simple, community-based framework for reproducibility.

Reviewer: Might need to wordsmith this a bit more.
```suggestion
in an end-to-end manner from the process of data inquiry to the presentation of the analyzed data,
and provide a simple, community-based framework for reproducibility.
```

Reviewer 2: I like this change

> throughout the data inquiry to analyzed data presentation process and provide a simple, community-based framework for reproducibility.

Author: I tried to incorporate all the ideas. Happy to make additional changes.

Open Science Example Use Cases
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Research is the primary driver for development of icepyx functionality.

> If you are or plan to work on a project using ICESat-2 data, we encourage you to use icepyx as a framework for finding and processing your data, 

Reviewer: ```suggestion
If you plan to or are working on a project using ICESat-2 data, we encourage you to use icepyx as a framework for finding and processing your data, 
```

Reviewer 2: Why say 'if you plan or are working ..' when it could just say, 'We encourage you to use icebox as a framework for finding and processing your ICESat-2 data....' seems simpler

icepyx aims to continually reduce the need for researchers to rewrite "routine" code by enabling easy visualization
throughout the data inquiry to analyzed data presentation process and provide a simple, community-based framework for reproducibility.

Open Science Example Use Cases
------------------------------
We are currently partnering with multiple researchers conducting investigations using ICESat-2 datasets. Their research, from data collection and analysis to publication, will be used to drive the development of icepyx functionality, ultimately providing software contributions and example workflows. Current collaborations focus on:
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Research is the primary driver for development of icepyx functionality.
If you are or plan to work on a project using ICESat-2 data, we encourage you to use icepyx as a framework for finding and processing your data, 
from designing your analysis to writing code to analyze your data to generating presentation-quality figures.
We welcome example use cases from all disciplines.

- impacts of blowing snow and low clouds on ICESat-2 measurements
- snow height in non-glaciated regions

> - parameter assimilation into sea ice models

Reviewer: To better resonate with people (especially non-ice ones), it might be good to have a few examples of disciplines. E.g. glaciology, coastal geomorphology, vegetation biomass, etc. Maybe a quick search on what ICESat-2 has been used for will help out with this list.

Reviewer 2: Agreed and think adding 'vegetation canopy height,' 'subsurface ocean structure,' 'sea level change' as suggestions

Author: So you would suggest NOT removing this? This original list was actually based on a non-funded proposal where specific groups had agreed to contribute examples as collaborators. Currently our examples center on "how to use icepyx" rather than research use cases. This would be good motivation for finishing out the pangeo-gallery steps to have a space for use case examples.

Reviewer: > So you would suggest NOT removing this? This original list was actually based on a non-funded proposal where specific groups had agreed to contribute examples as collaborators.

Yes, don't remove the list, but do modify it to include newer stuff that people have used ICESat-2 for. My guess it that people want to see something that *clicks* with what they're doing.

> Currently our examples center on "how to use icepyx" rather than research use cases. This would be good motivation for finishing out the pangeo-gallery steps to have a space for use case examples.

Yeah, really need to reboot https://github.com/icesat2py/icepyx-gallery. I feel that there's been lots of improvements since 2020 to the whole Binder/Sphinx/Docker/NSIDC-auth ecosystem, and it's time to get that up and running again. Certainly could look at https://github.com/ICESAT-2HackWeek/website2022 and copy some of the techniques from there.

Author: Re-added a list of example topics here to show the disciplinary breadth.

> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Reviewer: ```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

> The complexity of multiple data access systems, many with different metadata formats and API access types, 

Reviewer: ```suggestion
The complexity of multiple data access systems, often with different metadata formats and API access types, 
```

presents a challenge for finding and integrating diverse datasets. 
Driven by researcher use cases, icepyx contains a framework for easily adding a new product/sensor to an existing data analysis pipeline,

> improving researcher ability to easily compare diverse datasets across varying sensor types and spatial and temporal scales.

Reviewer: Debating on whether to take out the 'easily' adjective, when has data integration ever been easy :slightly_smiling_face:

Reviewer 2: maybe instead  '...icepyx contains a consistent framework for adding a new product/sensor.....' because I think the key here is that icepyx can enable downloads of many types of data, and it's easy because it's consistent and in one place.

Modifying the Development Plan
------------------------------
Everyone is invited to review and propose new themes for the Development Plan.
icepyx is continually evolving and its direction is driven by your feedback and contributions.

If you'd like to add a theme to this development plan, please submit your idea as a GitHub issue to solicit community feedback.

> Once there is agreement on your idea, submit a pull request to update the Development Plan, including a link to the discussion issue.

Reviewer: Wondering if there's value with enabling the 'Discussions' tab and migrating some of the https://github.com/icesat2py/icepyx/labels/Discussion%20-%20feedback%20solicited issues into there?

On one hand, it's yet another thing to manage, but on the other hand, we could separate long-winded discussion points from issues that are usually of a more short-term and technical nature (e.g. bug reports and feature requests).

Author: I've gone back and forth on this since we were using Pangeo's Discourse. I think if we enable discussions (which didn't exist then) we'd want to direct everything there, which could actually be good for maintenance because it's all in one place (and as you say, separates discussion from to-dos better).

Reviewer: Hmm, or we could just provide a link to https://discourse.pangeo.io/c/science/icesat-2/16? To be honest, the discourse forum isn't very active, and I'm not sure if using GitHub discussions would magically create more discussions. Leave it up to you to make that call.

Author: Discussions activated and discussion issues moved. We'll have to add a note and link to the contact us page.
